File: ktstrings/src/test/java/com/javaground/strings/anagram/countanagrams/CountAnagramsTest.java
package com.javaground.strings.anagram.countanagrams;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.*;

class CountAnagramsTest {

    public static Stream<Arguments> testCases() {
        return Stream.of(
                Arguments.of("too hot", 18),
                Arguments.of("aa", 1),
                Arguments.of("all good", 36),
                Arguments.of("a a a b b", 1),
                Arguments.of("hello world", 7200),
                Arguments.of("excel", 60),
                Arguments.of("ab ab cd cd ef ef", 64)
        );
    }

    @ParameterizedTest(name = "{index} => CountAnagrams.of({0}) should return {1}")
    @MethodSource("testCases")
    void countAnagrams(String s, int expected) {
        int actual = CountAnagrams.of(s);
        assertEquals(expected, actual);
    }
}

File: ktstrings/src/test/kotlin/com/kotlinground/strings/anagram/countanagrams/CountAnagramsTest.kt
package com.kotlinground.strings.anagram.countanagrams

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

class CountAnagramsTest {
    companion object {
        @JvmStatic
        fun testCases() = listOf(
            Arguments.of("too hot", 18),
            Arguments.of("aa", 1),
            Arguments.of("all good", 36),
            Arguments.of("a a a b b", 1),
            Arguments.of("hello world", 7200),
            Arguments.of("excel", 60),
            Arguments.of("ab ab cd cd ef ef", 64)
        )
    }

    @ParameterizedTest(name = "{index} => countAnagrams({0}) should return {1}")
    @MethodSource("testCases")
    fun countAnagrams(s: String, expected: Int) {
        val actual = countAnagrams(s)
        Assertions.assertEquals(expected, actual)
    }
}
